src/target/at32f43x.c

```diff
@@ -364,3 +393,155 @@ static bool at32f43_mass_erase(target_s *target)
 		return at32f43_mass_erase_bank(target, AT32F43x_FLASH_BANK2_REG_OFFSET, &timeout);
 	return true;
 }
+
+static bool at32f43_option_erase(target_s *target)
+{
+	/* bank_reg_offset is 0, option bytes belong to first bank */
+	at32f43_flash_clear_eop(target, 0);
+	DEBUG_TARGET("%s\n", __func__);
+
+	/* Wipe User System Data */
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL, AT32F43x_FLASH_CTRL_USDERS | AT32F43x_FLASH_CTRL_USDULKS);
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL,
+		AT32F43x_FLASH_CTRL_USDERS | AT32F43x_FLASH_CTRL_USDULKS | AT32F43x_FLASH_CTRL_ERSTR);
+
+	return at32f43_flash_busy_wait(target, 0, NULL);
+}
+
+static bool at32f43_option_write_erased(target_s *const target, const size_t offset, const uint16_t value)
+{
+	if (value == 0xffffU)
+		return true;
+
+	at32f43_flash_clear_eop(target, 0);
+
+	/* Enable writing User System Data */
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL, AT32F43x_FLASH_CTRL_USDPRGM | AT32F43x_FLASH_CTRL_USDULKS);
+
+	const uint32_t addr = AT32F43x_USD_BASE + (offset * 2U);
+	DEBUG_TARGET("%s: 0x%08X <- 0x%04X\n", __func__, addr, value);
```

Comment:
`addr` is a uint32_t, so please use `PRIX32` or `PRIx32` for it here.

Reply:
Yes, thanks. Verified with enabled DEBUG_TARGET, it used to fail the firmware build (-Wformat). Fixed.